### PR TITLE
CLI: Fix default nonce authority resolution

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

If the nonce authority was not explicitly provided, the nonce account itself was erroneously assigned authority rather than the account creator

#### Summary of Changes

Resolve unspecified nonce authority to the nonce account's creator

### CODE REVIEW

File: cli/src/nonce.rs [hunk 1]
```diff
@@ -231,8 +226,8 @@ pub fn parse_authorize_nonce_account(matches: &ArgMatches<'_>) -> Result<CliComm
 pub fn parse_nonce_create_account(matches: &ArgMatches<'_>) -> Result<CliCommandInfo, CliError> {
     let nonce_account = keypair_of(matches, "nonce_account_keypair").unwrap();
     let lamports = required_lamports_from(matches, "amount", "unit")?;
-    let nonce_authority =
-        pubkey_of(matches, "nonce_authority").unwrap_or_else(|| nonce_account.pubkey());
+    let nonce_authority = pubkey_of(matches, "nonce_authority");
+    println!("na: {:?}", nonce_authority);
```
Comment (jstarry) [inline cli/src/nonce.rs]: nit: remove

File: cli/tests/nonce.rs [hunk 2]
```diff
@@ -122,7 +122,7 @@ fn full_battery_tests(
     // Create nonce account
     config_payer.command = CliCommand::CreateNonceAccount {
         nonce_account: read_keypair_file(&nonce_keypair_file).unwrap().into(),
-        nonce_authority: read_keypair_file(&authority_keypair_file).unwrap().pubkey(),
+        nonce_authority: Some(read_keypair_file(&authority_keypair_file).unwrap().pubkey()),
```
Comment (jstarry) [inline cli/tests/nonce.rs]: I think there should be integration tests for the implied nonce authority case. `nonce_authority: None`, etc.

File: cli/src/nonce.rs (hunk 1 above)
Comment (t-nelson) [reply cli/src/nonce.rs]: Whoops! 🤦‍♂

File: cli/tests/nonce.rs (hunk 2 above)
Comment (t-nelson) [reply cli/tests/nonce.rs]: Hrm.  Yeah, I'll fix up the non-authority case to call through with `None` instead of the same keypair

**Pull request merged**